tests/models/test_work.py

```diff
@@ -491,30 +491,55 @@ def test_set_presentation_ready(self):
 
         # But the work of adding it to the search engine has been
         # registered.
-        [record] = [
-            x for x in work.coverage_records
-            if x.operation==WorkCoverageRecord.UPDATE_SEARCH_INDEX_OPERATION
-        ]
-        eq_(WorkCoverageRecord.REGISTERED, record.status)
-
-        # This work is presentation ready because it has a title
+        def assert_record():
+            # Verify the search index WorkCoverageRecord for this work
+            # is in the REGISTERED state.
+            [record] = [
+                x for x in work.coverage_records
+                if x.operation==WorkCoverageRecord.UPDATE_SEARCH_INDEX_OPERATION
+            ]
+            eq_(WorkCoverageRecord.REGISTERED, record.status)
+        assert_record()
 
+        # This work is presentation ready because it has a title.
         # Remove the title, and the work stops being presentation
         # ready.
         presentation.title = None
         work.set_presentation_ready_based_on_content(search_index_client=search)
         eq_(False, work.presentation_ready)
 
-        # The work has been removed from the search index.
-        eq_([], search.docs.keys())
```

Comment:
The behavior didn't change, but this assertion stopped having any value when we switched from "immediately remove an unready work from the search index" to "set this work's WorkCoverageRecord back to the REGISTERED state so the search index can handle it later." The thing we actually want to test is `assert_record()`.

Reply:
This is related to the comment you made about how the search code is not relevant in `set_presentation_ready_based_on_content` anymore, right?

Reply:
Yes, exactly.